lib/train/transports/local.rb

```diff
@@ -21,13 +21,19 @@ def initialize(options)
         super(options)
         @cmd_wrapper = nil
         @cmd_wrapper = CommandWrapper.load(self, options)
+        @platform = platform
       end
 
       def run_command(cmd)
-        cmd = @cmd_wrapper.run(cmd) unless @cmd_wrapper.nil?
-        res = Mixlib::ShellOut.new(cmd)
-        res.run_command
-        CommandResult.new(res.stdout, res.stderr, res.exitstatus)
+        if defined?(@platform) && @platform.windows?
+          @windows_runner ||= WindowsRunner.new
```

Comment:
The `WindowsRunner` is a great step in the right direction. Should we do this analog to our file handling? eg. 
```
Train::Command::Local::UnixRunner.new()
Train::Command::Local::WindowsPipeRunner.new()
# Fallback when the pipe is not working
Train::Command::Local::WindowsShellRunner.new()
```

Reply:
I like the idea of separating the Windows runners. I don't see a reason for a `UnixRunner` at this time. I may create a `GenericRunner` which will use `ShellOut` and be used by Linux and the first few commands that determine the OS.

Reply:
The unixrunner was the default shell out :-)

Reply:
Yup, but a few commands are ran using ShellOut while the OS is being identified, that's why I opted for `GenericRunner`.